YSE_App/models/transient_models.py

```diff
@@ -21,329 +21,359 @@
-		dec_offset = cd.Angle(15, unit=u.deg)
-
-		sf = SurveyField.objects.filter(~Q(obs_group__name='ZTF')).\
-			filter((Q(ra_cen__gt = self.ra-ra_offset.degree) &
-					Q(ra_cen__lt = self.ra+ra_offset.degree) &
-					Q(dec_cen__gt = self.dec-dec_offset.degree) &
-					Q(dec_cen__lt = self.dec+dec_offset.degree))).select_related()
-
-		sc = cd.SkyCoord(self.ra,self.dec,unit=u.deg)
-		sc2 = cd.SkyCoord(sf[0].ra_cen,sf[0].dec_cen,unit=u.deg)
-		if len(sf): 
-			return '%.1f'%sc.separation(sc2).degree
-		else: return None
-		
-		
-	def modified_date_pacific(self):
-		date_format = '%m/%d/%Y %H:%M:%S %Z'
-		mod_date = self.modified_date.astimezone(timezone('US/Pacific'))
-		return mod_date.strftime(date_format)
-
-	def disc_date_string(self):
-		date_format = '%m/%d/%Y'
-		return self.disc_date.strftime(date_format)
-
-	def disc_mag(self):
-
-		transient_query = Q(transient=self.id)
-		all_phot = yse_models.TransientPhotometry.objects.filter(transient_query)
-		phot_ids = all_phot.values('id')
-		phot_data_query = Q(photometry__id__in=phot_ids)
-		disc_query = Q(discovery_point = 1)
-		disc_mag = yse_models.TransientPhotData.objects.exclude(data_quality__isnull=False).filter(phot_data_query & disc_query)
-		if len(disc_mag):
-			return disc_mag[0].mag
-		else:
-			return None
-
-	def recent_mag(self):
-		date_format = '%m/%d/%Y'
-
-		transient_query = Q(transient=self.id)
-		all_phot = yse_models.TransientPhotometry.objects.filter(transient_query)
-		phot_ids = all_phot.values('id')
-		phot_data_query = Q(photometry__id__in=phot_ids)
-		recent_mag = yse_models.TransientPhotData.objects.exclude(data_quality__isnull=False).filter(mag__isnull=False).filter(phot_data_query).order_by('-obs_date')
-
-		if len(recent_mag):
-			return '%.2f'%(recent_mag[0].mag)
-		else:
-			return None
-
-	def recent_magdate(self):
-		date_format = '%m/%d/%Y'
-
-		transient_query = Q(transient=self.id)
-		all_phot = yse_models.TransientPhotometry.objects.filter(transient_query)
-		phot_ids = all_phot.values('id')
-		phot_data_query = Q(photometry__id__in=phot_ids)
-		recent_mag = yse_models.TransientPhotData.objects.exclude(data_quality__isnull=False).filter(phot_data_query).order_by('-obs_date')
-
-		if len(recent_mag):
-			return '%s'%(recent_mag[0].obs_date.strftime(date_format))
-		else:
-			return None
-
-	def z_or_hostz(self):
-		if self.redshift:
-			return self.redshift
-		elif self.host and self.host.redshift:
-			return self.host.redshift
-		else: return None
-
-	def name_table_sort(self):
-		if len(self.name) > 4:
-			addnums = 7-len(self.name)
-			sortname = "".join([self.name[:4],"".join(['1']*addnums),self.name[4:]])
-
-			return sortname
-		else:
-			return None
-
-	def __str__(self):
-		return self.name
-
-	def natural_key(self):
-		return self.name
+    has_hst = models.BooleanField(null=True, blank=True)
+    has_spitzer = models.BooleanField(null=True, blank=True)
+    has_chandra = models.BooleanField(null=True, blank=True)
```

Comment:
Also check for `has_vlass`? e.g., via CIRADA?

Reply:
good ideas;  would have to implement